src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String OPERATION_TYPE_BUY = "buy";
+    private static final String OPERATION_TYPE_SUPPLY = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> records = new ArrayList<>();
+        StringBuilder report = new StringBuilder();
+        readFromFile(fromFileName, records, report);
+        writeToFile(toFileName, report);
+    }
+
+    private static void readFromFile(String fromFileName,
+                                     List<String> stringFromFile, StringBuilder report) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String read = bufferedReader.readLine();
+            while (read != null) {
+                stringFromFile.add(read);
+                read = bufferedReader.readLine();
+            }
+            int buy = 0;
+            int supply = 0;
+
+            for (String s : stringFromFile) {
+                String[] split = s.split(",");
+                int digit = Integer.parseInt(split[1]);
+                if (split[0].equals(OPERATION_TYPE_BUY)) {
+                    buy += digit;
+                } else if (split[0].equals(OPERATION_TYPE_SUPPLY)) {
+                    supply += digit;
+                }
```

Comment:
Instead of using 'split' as the variable name, it would be more informative to name it something like 'recordParts' or 'transactionDetails' to reflect its contents.